friends/friendController.js

```diff
@@ -0,0 +1,88 @@
+const router = require('express').Router();
+
+const Friend = require('./friendSchema');
+
+router.route('/')
+    .get((req, res) => {
+        Friend
+            .find()
+            .then(friends => {
+                res.status(200).json(friends);
+            })
+            .catch(err => {
+                res.status(500).json({ error: "The friends information could not be retrieved." })
+            })
+    })
+    .post((req, res) => {
+        const friendData = req.body;
+        const friend = new Friend(friendData);
+        // console.log(req.body);
+        if (!req.body.firstName || !req.body.lastName || !req.body.age) {
```

Comment:
This is a good opportunity to use object destructuring:

```
const { age, firstName, lastName } req.body

if (!age || !firstName || !lastName)
  return res.status(400).json({ error: "firstName, lastName, and age are required!"})
```

Reply:
That is really weird; how does JavaScript know what to do with variables age, firstName, and lastName? What if we were to put a variable that didn't exist inside those curly brackets?

Reply:
As Frank said, destructuring in JS is a sort of shorthand. Because the `{}`'s are around the variables, JS expects there to be an object with keys `age`, `firstName`, and `lastName`, so then you don't need `req.body` in front of each variable on line 20. If you put a variable that didn't exist, I would expect an error to be thrown, but please try that out! :)